File: cpp/bc1164.cpp
#include <iostream>

using namespace std;

int main()
{
	int count, val, sum;
	string msg = "";
	
	cin >> count;
	
	for(int i = 0; i < count; i++)
	{
		cin >> val;
		sum = 0;
		
		for(int j = 1; j < val; j++)
			if(val % j == 0) sum += j;
		
		if(sum == val) msg = "eh perfeito";
		else msg = "nao eh perfeito";
		
		cout << val << " " << msg << endl;
	}
}

File: python/bc1164.py
count = int(input())

for c in range(0, count):
    su = 0
    val = int(input())

    for d in range(1, val):
        if val % d == 0: su += d

    if su == val:
        print(f"{val} eh perfeito")
    else:
        print(f"{val} nao eh perfeito")